### PR TITLE
add graph ql and create middleware for it

### PR DESCRIPTION
**Pull request by jennie-quinton**
I made my own middleware

### CODE REVIEW

File: client/middleware/graphQL.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import {graphql} from 'graphql';
+
+var root = {
```
Comment (jennie-quinton) [inline client/middleware/graphQL.js:3]: to be replaced with the call to the backend

File: client/pages/index.js [hunk 2]
```diff
@@ -25,8 +25,10 @@ const mapStateToProps = (state ) => {
   return {};
 }
 
-const mapDispatchToProps =(dispatch)=>  {
-  getInitialCharacter: dispatch(addCharacter())
-}
+const mapDispatchToProps = (dispatch)=> {
```
Comment (jordanrios94) [inline client/pages/index.js:28]: probs a bit of es6 refactoring if you want

```
const mapDispatchToProps = dispatch => ({
  getInitialCharacter: () => dispatch(addCharacter())
})
```

File: client/reducers/characterReducer.js [hunk 3]
```diff
@@ -8,7 +6,7 @@
     const newAction = {...action};
 
     switch(action.type) {
-      case 'CREATE_CHARACTER': {
+      case 'CREATE_CHARACTER_SUCCESS': {
```
Comment (jordanrios94) [inline client/reducers/characterReducer.js:9]: why did you change it to `CREATE_CHARACTER_SUCCESS `?
does the action need to update the type as well?

File: client/webpack.config.js [hunk 4]
```diff
@@ -11,6 +11,12 @@ module.exports = {
   },
   module: {
     rules: [
+      // needed for graphql
+      {
+        test: /\.mjs$/,
```
Comment (jordanrios94) [inline client/webpack.config.js:16]: nice :)

File: client/reducers/characterReducer.js (hunk 3 above)
Comment (jordanrios94) [reply client/reducers/characterReducer.js:9]: I see now how it gets set in `client/middleware/graphQL.js`

File: client/middleware/graphQL.js (hunk 1 above)
Comment (jordanrios94) [reply client/middleware/graphQL.js:3]: I can see we are using `CREATE_CHARACTER`, surely that will be of type `mutation` rather than 'query' as we are creating a new record or are just reading for now?

**Comment by jordanrios94** [comment]
So bearing in mind we will be posting to `/graphql`, and the file structure is looking good. We can merge this if you are happy with it as well. As long as in your code we change between types of `query` and `mutation` I am happy with it.

p.s. on the server I will expect the following to be posted. Just so I know we are the same page.

```
"{"query":"{ hello }"}"
```

or

```
"{"query":"mutation {\n addUser(firstName: \"Stephen\", age: 26) ... ""
```

We don't have a graphql server to test with, therefore I will go ahead and create one.

File: client/actions/schemas/characterSchema.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import {buildSchema} from 'graphql';
```
Comment (jordanrios94) [inline client/actions/schemas/characterSchema.js:1]: this file will be duplicated for the backend, therefore we will need to move to this top level when I start on the backend

File: client/actions/characterActions.js [hunk 6]
```diff
@@ -1,12 +1,11 @@
...
   return dispatch({
     type: 'CREATE_CHARACTER',
-    data: {
-      name: 'Mikely Gregor',
-      race: 'Human',
-      class: 'Monk',
-    }
-  })
+    schema: characterSchema,
+    query: '{ name, race, class }'
```
Comment (jordanrios94) [inline client/actions/characterActions.js:7]: is the schema necessary for the FE? This is definitely needed on the BE as there is where the graph will be created. FE from my point of just needs to POST the query. I could however be wrong by the way.

File: client/middleware/graphQL.js (hunk 1 above)
Comment (jennie-quinton) [reply client/middleware/graphQL.js:3]: just reading, sorry names of functions are wrong atm there

File: client/pages/index.js (hunk 2 above)
Comment (jennie-quinton) [reply client/pages/index.js:28]: I am using a linter it does it for me
I can change lint rules if u wish

File: client/reducers/characterReducer.js (hunk 3 above)
Comment (jennie-quinton) [reply client/reducers/characterReducer.js:9]: :D

File: client/actions/schemas/characterSchema.js (hunk 5 above)
Comment (jennie-quinton) [reply client/actions/schemas/characterSchema.js:1]: awesome

File: client/actions/characterActions.js (hunk 6 above)
Comment (jennie-quinton) [reply client/actions/characterActions.js:7]: sorry I have naming off here, this is get character

Comment (jennie-quinton) [reply client/actions/characterActions.js:7]: however, from my understanding that query line is what i specify to say the information i want

File: client/middleware/graphQL.js (hunk 1 above)
Comment (jennie-quinton) [reply client/middleware/graphQL.js:3]: and i havent looked into mutation, this is just the very basics of adding graph ql
